File: app/src/main/java/com/hhy/game/snowman/game/ScoreText.java
package com.hhy.game.snowman.game;

import android.content.res.Resources;
import android.graphics.Bitmap;
import android.graphics.BitmapFactory;
import android.graphics.Canvas;
import android.graphics.Paint;
import android.graphics.Typeface;
import com.hhy.game.snowman.R;

class ScoreText {
    private final Paint drawingPaint = new Paint(Paint.ANTI_ALIAS_FLAG);
    private final Paint scoresPaint = new Paint(Paint.ANTI_ALIAS_FLAG);
    private final Paint recordPaint = new Paint(Paint.ANTI_ALIAS_FLAG);
    private final Paint pauseTextPaint = new Paint(Paint.ANTI_ALIAS_FLAG);

    private static Bitmap recordBitmap;
    private Values values;
    private Resources resources;

    ScoreText(Values values, Resources resources) {
        this.values = values;
        this.resources = resources;

        scoresPaint.setColor(resources.getColor(R.color.colorText));
        scoresPaint.setTextSize(values.getScoreTextSize());
        scoresPaint.setTypeface(Typeface.SERIF);
        recordPaint.setColor(resources.getColor(R.color.colorText));
        recordPaint.setTextSize(values.getRecordTextSize());
        recordPaint.setTypeface(Typeface.SERIF);
        pauseTextPaint.setColor(resources.getColor(R.color.colorText));
        pauseTextPaint.setTextSize(values.getPausedTextSize());
        pauseTextPaint.setTypeface(Typeface.SERIF);

        recordBitmap = Bitmap.createScaledBitmap(BitmapFactory.decodeResource(resources, R.drawable.records),
                values.getRecordImgWidth(),
                values.getRecordImgHeight(), false);
    }

    void drawScore(Canvas canvas, String text) {
        canvas.drawText(text, values.getScoreTextX(), values.getScoreTextY() + scoresPaint.getTextSize(), scoresPaint);
    }
    void drawRecord(Canvas canvas, String text) {
        canvas.drawBitmap(recordBitmap, values.getRecordX(), values.getRecordY(), drawingPaint);
        canvas.drawText(text, values.getRecordX() + recordBitmap.getWidth(),
                values.getRecordY() + recordPaint.getTextSize(), recordPaint);
    }
    void drawPause(Canvas canvas) {
        canvas.drawText(resources.getString(R.string.pause), values.getPausedTextX(), values.getPausedTextY(), pauseTextPaint);
    }
}

File: app/src/main/java/com/hhy/game/snowman/game/ScoreText.kt
package com.hhy.game.snowman.game

import android.content.Context
import android.content.res.Resources
import android.graphics.Bitmap
import android.graphics.BitmapFactory
import android.graphics.Canvas
import android.graphics.Paint
import android.graphics.Typeface
import androidx.core.content.ContextCompat
import com.hhy.game.snowman.R

internal class ScoreText(private val context: Context, private val values: Values, private val resources: Resources) {
    private lateinit var recordBitmap: Bitmap
    private val drawingPaint = Paint(Paint.ANTI_ALIAS_FLAG)
    private val scoresPaint = Paint(Paint.ANTI_ALIAS_FLAG)
    private val recordPaint = Paint(Paint.ANTI_ALIAS_FLAG)
    private val pauseTextPaint = Paint(Paint.ANTI_ALIAS_FLAG)

    // draw score
    fun drawScore(canvas: Canvas, text: String) {
        canvas.drawText(text, values.scoreTextX.toFloat(), values.scoreTextY + scoresPaint.textSize, scoresPaint)
    }

    // draw record
    fun drawRecord(canvas: Canvas, text: String) {
        canvas.drawBitmap(recordBitmap, values.recordX.toFloat(), values.recordY.toFloat(), drawingPaint)
        canvas.drawText(text, values.recordX + recordBitmap.width.toFloat(),
            values.recordY + recordPaint.textSize, recordPaint)
    }

    // draw paused text
    fun drawPause(canvas: Canvas) {
        canvas.drawText(resources.getString(R.string.pause), values.pausedTextX.toFloat(), values.pausedTextY.toFloat(), pauseTextPaint)
    }

    init {
        // score text
        scoresPaint.color = ContextCompat.getColor(context, R.color.colorText)
        scoresPaint.textSize = values.scoreTextSize.toFloat()
        scoresPaint.typeface = Typeface.SERIF
        // record text
        recordPaint.color = ContextCompat.getColor(context, R.color.colorText)
        recordPaint.textSize = values.recordTextSize.toFloat()
        recordPaint.typeface = Typeface.SERIF
        // paused text
        pauseTextPaint.color = ContextCompat.getColor(context, R.color.colorText)
        pauseTextPaint.textSize = values.pausedTextSize.toFloat()
        pauseTextPaint.typeface = Typeface.SERIF

        // record image
        recordBitmap = Bitmap.createScaledBitmap(BitmapFactory.decodeResource(resources, R.drawable.records),
            values.recordImgWidth,
            values.recordImgHeight, false)
    }
}
